app/src/main/java/com/example/criminalintent/di/AppModule.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.criminalintent.di
+
+import androidx.room.Room
+import androidx.room.migration.Migration
+import androidx.sqlite.db.SupportSQLiteDatabase
+import com.example.criminalintent.data.local.CrimeDatabase
+import com.example.criminalintent.data.repository.CrimeRepository
+import com.example.criminalintent.presentation.auth.login.LoginViewModel
+import com.example.criminalintent.presentation.auth.signup.SignUpViewModel
+import com.example.criminalintent.presentation.crimeDetail.CrimeDetailViewModel
+import com.example.criminalintent.presentation.crimeList.CrimeListViewModel
+import org.koin.android.ext.koin.androidApplication
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.dsl.module
+
+val appModule = module {
+    single {
+        Room.databaseBuilder(
+            androidApplication(),
+            CrimeDatabase::class.java,
+            CrimeRepository.DATABASE_NAME
+        )
+            .addMigrations(migration()).build().crimeDao()
+    }
+    single {
```

Comment:
Dependency injection provides some benefits such as: replacing implementations and eases testability. In order to benefit from those, you have to inject an interface instead of a concrete class.
Please define interfaces, e.g CrimeRepository and concrete implementations e.g CrimeRepositoryImpl